src/main/java/bio/terra/grammar/google/BigQueryVisitor.java

```diff
@@ -16,6 +17,12 @@ public BigQueryVisitor(Map<String, DatasetModel> datasetMap) {
   }
 
   public String generateAlias(String datasetName, String tableName) {
+    if (datasetName == null || tableName == null) {
```

Comment:
I don't think it's possible for `datasetName` to be `null`. It's possible for `ctx.dataset_name()` to be `null` but since the name is always prefixed by `prefixDatasetName()` before it gets here, it will never be `null`. If it's possible that it is `null` in practice, we need to do this check in the caller before the prefix is added.

If this check is done in the caller, then we have `Table_exprContext` or `Column_exprContext` which would let us include the specific part of the query in the error message, which we could use to make a more helpful error message.

Reply:
Great catch, thank you so much!
It turns out that we don't even let an unqualified table name even get to the translate step: it fails on parsing the query. So, I've moved the logic to the column call and removed the check for the dataset. I've added a test for parsing an unqualified dataset, so we should be alerted if we try to update the grammar to no longer require qualified table names.